ramp-engine/ramp_engine/aws/api.py

```diff
@@ -55,6 +55,10 @@
 TRAIN_LOOP_INTERVAL_SECS_FIELD = 'train_loop_interval_secs'
 MEMORY_PROFILING_FIELD = 'memory_profiling'
 
+# how long to wait for connections
+WAIT_MINUTES = 10
+MAX_TRIES_TO_CONNECT = 6
```

Comment:
1hour waiting time seems a lot no?
But I have no idea how frequent the failures are?

Reply:
Thanks for your comments :-) @tomMoral
Yes. 1h seems like a lot but I think in my particular case (when there are only 3 instances available) It's good to give the submission a bit of a chance to get submitted: otherwise it will fail and the error will be coming from not having instances, not the wrong submission.
I think it would be better to put those two params in the config file of the event in the future, but for now I prefer to keep it as is so the events (especially the locomotion challenge) would not have to be redeployed

Reply:
> when there are only 3 instances available)

It might be worth following up with Nicolas about this. Also requesting for increase in spot instance limit seems to be much easier than requesting for increase in on-demand instance, I think if you requested to increase spot instance limit it should go through

Reply:
I requested in spot instance limit -> the use of on demand is now broken on the current release of RAMP anyways
(I made the PR to fix it: https://github.com/paris-saclay-cds/ramp-board/pull/462).